### PR TITLE
Sample project

### PR DESCRIPTION
**Pull request by bruno-garcia**
I followed [the ideas on this post](https://medium.com/@charpeni/setting-up-an-example-app-for-your-react-native-library-d940c5cf31e4) on how to add a sample project in the library while being able to change the lib and see the result in the app.

It worked on Android but I'm having a hard time with xcode.
Opening this draft in case it's obvious to anyone.

TODO:
 - [x] Fix `xcworkspace` to build `sample` as well as `@sentry/react-native` so that changes to the native code shows up on the sample app.
 - [ ] Change UI to have some sample buttons

### CODE REVIEW

File: sample/.buckconfig [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
+[android]
```
Comment (marandaneto) [inline sample/.buckconfig:2]: if we use `Gradle` for Android, we don't need `Buck`

File: sample/android/app/_BUCK [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# To learn about Buck see [Docs](https://buckbuild.com/).
```
Comment (marandaneto) [inline sample/android/app/_BUCK:1]: same as https://github.com/getsentry/sentry-react-native/pull/837/files#r422104024

File: sample/.buckconfig (hunk 1 above)
Comment (bruno-garcia) [reply sample/.buckconfig:2]: This came up with the react-native init project

File: sample/android/app/build.gradle [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+
+    debugImplementation("com.facebook.flipper:flipper:${FLIPPER_VERSION}") {
+      exclude group:'com.facebook.fbjni'
+    }
+
+    debugImplementation("com.facebook.flipper:flipper-network-plugin:${FLIPPER_VERSION}") {
+        exclude group:'com.facebook.flipper'
+    }
+
+    debugImplementation("com.facebook.flipper:flipper-fresco-plugin:${FLIPPER_VERSION}") {
```
Comment (marandaneto) [inline sample/android/app/build.gradle:207]: guess this won't be any useful, fresco is an image loading lib., we dont need to debug this

File: sample/android/app/_BUCK (hunk 2 above)
Comment (bruno-garcia) [reply sample/android/app/_BUCK:1]: Could be an improvement but I'd like to land the working sample in first.
Android actually works already

File: sample/android/app/build.gradle (hunk 3 above)
Comment (bruno-garcia) [reply sample/android/app/build.gradle:207]: I believe this is used to hot reload sstuff, it seems this flipper is all over. I'm not sure yet, it's just the template sample.

File: sample/.buckconfig (hunk 1 above)
Comment (marandaneto) [reply sample/.buckconfig:2]: yes I know, just afraid to make the sample heavier and even harder to set up

File: sample/android/app/build.gradle (hunk 3 above)
Comment (marandaneto) [reply sample/android/app/build.gradle:207]: flipper it is, but not all the modules of flipper, eg the `flipper-network-plugin` makes sense, but not a library loader debugger

File: sample/.buckconfig (hunk 1 above)
Comment (bruno-garcia) [reply sample/.buckconfig:2]: I'm afraid of spending more hours "refactoring" this. I'd like to get this working first. We can remove all unecessary code after. I'd focus on adding cocoa 5.0 before that.

File: sample/android/app/build.gradle (hunk 3 above)
Comment (marandaneto) [reply sample/android/app/build.gradle:207]: yes yes, but we could clean it up, that's what I mean, happy to merge as it is

Comment (bruno-garcia) [reply sample/android/app/build.gradle:207]: Thanks

**Comment by HazAT** [comment]
Thank you @bruno-garcia

**Pull request merged**